sklearn/decomposition/tests/test_pca.py

```diff
@@ -340,11 +341,29 @@ def test_pca_inverse():
 
 
 def test_pca_validation():
-    X = [[0, 1], [1, 0]]
+    # Ensures that extreme inputs for n_components common to all solvers
+    # (less than 0 or more than the lesser dimension of the input
+    # matrix X) raise errors.
+    X = np.array([[0, 1, 0], [1, 0, 0]])
     for solver in solver_list:
         for n_components in [-1, 3]:
-            assert_raises(ValueError,
-                          PCA(n_components, svd_solver=solver).fit, X)
+            assert_raises_regex(ValueError,
```

Comment:
X.T: I can add it alright.

Boundary case: the shape of X is (2,3); so for the "full" solver the maximum number of components is 2, while for "arpack" it is 1. Regarding "I'm not very familiar with this code / solvers", I'm not sure what you mean. I actually just extended the non-regression test that was already present for my fix, so the original test may not be best practice. It looked good and functional to me.